### DISCUSSION TITLE
githooks don't work automatically

### DISCUSSION THREAD

**Discussion by timdeschryver** [community/community#30820] [Codespaces]
Hi, first of all I want to tell that I like to use codespaces.
After using it for a few weeks for my project, I bumped against one issue and Craig asked me to create a discussion here.

The project is using githooks to perform a few checks before I can commit my files.

It was here that I encountered the following problem (in VS Code and in the browser):
(I think this was when I was using the default devcontainer)

```
hint: The '.githooks/pre-commit' hook was ignored because it's not set as executable.
```

The fix for it was to change the access permissions on the githooks folder:

```
chmod ug+x .githooks
```

Afterward, the error disappeared (even after reloading the workspace).

More information:

- Local OS: Windows 11
- Project: https://github.com/timdeschryver/timdeschryver.dev ([pre-commit githook](https://github.com/timdeschryver/timdeschryver.dev/blob/main/.githooks/pre-commit))

Thanks again!

**Comment by alexagillman** [comment]
Just to confirm, things are working as expected for you now? Thanks for bringing this to our attention

**Reply by timdeschryver** [reply]
Yes, after running the `chmod` command everything is working as expected.

**Reply by timdeschryver** [reply]
FWIW, In other repositories/codespaces it seems to be working without the need to execute the `chmod` command.
